clients/include/testing_copy_strided_batched.hpp

```diff
@@ -0,0 +1,186 @@
+/* ************************************************************************
+ * Copyright 2018-2019 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+
+#include "cblas_interface.hpp"
+#include "norm.hpp"
+#include "rocblas.hpp"
+#include "rocblas_init.hpp"
+#include "rocblas_math.hpp"
+#include "rocblas_random.hpp"
+#include "rocblas_test.hpp"
+#include "rocblas_vector.hpp"
+#include "unit.hpp"
+#include "utility.hpp"
+
+template <typename T>
+void testing_copy_strided_batched_bad_arg(const Arguments& arg)
+{
+    rocblas_int    N           = 100;
+    rocblas_int    incx        = 1;
+    rocblas_int    incy        = 1;
+    rocblas_stride stride_x    = incx * N;
+    rocblas_stride stride_y    = incy * N;
+    rocblas_int    batch_count = 5;
+
+    rocblas_local_handle handle;
+
+    size_t size_x = stride_x * batch_count;
+    size_t size_y = stride_y * batch_count;
+
+    device_vector<T> dx(size_x);
+    device_vector<T> dy(size_y);
+
+    if(!dx || !dy)
+    {
+        CHECK_HIP_ERROR(hipErrorOutOfMemory);
+        return;
+    }
+
+    EXPECT_ROCBLAS_STATUS(rocblas_copy_strided_batched<T>(
+                              handle, N, nullptr, incx, stride_x, dy, incy, stride_y, batch_count),
+                          rocblas_status_invalid_pointer);
+    EXPECT_ROCBLAS_STATUS(rocblas_copy_strided_batched<T>(
+                              handle, N, dx, incx, stride_x, nullptr, incy, stride_y, batch_count),
+                          rocblas_status_invalid_pointer);
+    EXPECT_ROCBLAS_STATUS(rocblas_copy_strided_batched<T>(
+                              nullptr, N, dx, incx, stride_x, dy, incy, stride_y, batch_count),
+                          rocblas_status_invalid_handle);
+}
+
+template <typename T>
+void testing_copy_strided_batched(const Arguments& arg)
+{
+    rocblas_int          N           = arg.N;
+    rocblas_int          incx        = arg.incx;
+    rocblas_int          incy        = arg.incy;
+    rocblas_int          stride_x    = arg.stride_x;
+    rocblas_int          stride_y    = arg.stride_y;
+    rocblas_int          batch_count = arg.batch_count;
+    rocblas_local_handle handle;
+    rocblas_int          abs_incx = incx >= 0 ? incx : -incx;
+    rocblas_int          abs_incy = incy >= 0 ? incy : -incy;
+    size_t               size_x   = N * size_t(abs_incx);
+    size_t               size_y   = N * size_t(abs_incy);
+
+    // argument sanity check before allocating invalid memory
+    if(N <= 0 || batch_count <= 0)
```

Comment:
`batch_count == 0` should probably follow normal path -- we want to make sure destination is not copied over